doit/dependency.py

```diff
@@ -361,6 +338,94 @@ def remove_all(self):
         self._conn.execute('delete from doit')
 
 
+class FileChangedChecker(object):
+    """Base checker for dependencies, must be inherited."""
+
+    def check_modified(self, file_path, state):
+        """Check if file in file_path is modified from previous "state".
+
+        @param file_path (string): file path
+        @param state (tuple): state that was previously saved with
+            ``get_state``
+        @returns (bool): True if dep is modified
+        """
+        pass
+
+    def get_state(self, task, dep, current_state):
+        """Compute the state of a task after it has been successfuly executed.
+
+        @param task (Task): the task object.
```

Comment:
Ah indeed, I think we can remove it, the dep file should be enough to compute the state.